integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java

```diff
@@ -103,148 +137,392 @@
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+    logger.info("verify the DomainValidationError event is generated");
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_VALIDATION_ERROR, "Warning", timestamp);
+
+    // remove the managed server from domain resource
+    timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
+    patchStr
+        = "[{\"op\": \"remove\",\"path\": \""
+        + "/spec/managedServers\""
+        + "}]";
+    logger.info("Updating domain configuration using patch string: {0}\n", patchStr);
+    patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    logger.info("verify the DomainProcessingCompleted event is generated");
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_CHANGED, "Normal", timestamp);
+  }
+
+  /**
+   * Change a domain resource with adding a cluster not in domain and verify the warning event is generated.
+   */
+  @Order(3)
+  @Test
+  @DisplayName("Test domain DomainValidationError event for non-existing cluster")
+  public void testDomainK8sEventsNonExistingCluster() {
+    DateTime timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
+    logger.info("patch the domain resource with new cluster");
+    String patchStr
+        = "["
+        + "{\"op\": \"add\",\"path\": \"/spec/clusters/-\", \"value\": "
+        + "    {\"clusterName\" : \"nonexisting-cluster\", \"replicas\": 2, \"serverStartState\": \"RUNNING\"}"
+        + "}]";
+    logger.info("Updating domain configuration using patch string: {0}\n", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+    // verify the DomainValidationError event is generated
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_VALIDATION_ERROR, "Warning", timestamp);
+
+    //remove the cluster from domain resource
+    timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
+    patchStr = "[{\"op\": \"remove\",\"path\": \"/spec/clusters/1\"}]";
+    logger.info("Updating domain configuration using patch string: {0}\n", patchStr);
+    patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    // verify the DomainProcessingStarting/Completed event is generated
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_CHANGED, "Normal", timestamp);
+  }
+
+  /**
+   * Test the following domain events are logged when domain resource goes through various life cycle stages.
    * Verifies DomainChanged is logged when operator processes the domain resource changes.
```

Comment:
Can you describe how the event is  triggered or what action triggers the event.  This is a generic review comments for all usecases.

Reply:
done